persistence/src/test/java/cz/muni/fi/pa165/mushrooms/dao/VisitDaoTest.java

```diff
@@ -151,9 +148,237 @@ public void create_validVisit() throws Exception {
         em.persist(testVisit);
 
         assertThat(visitDao.findAll().size()).isEqualTo(sizeBeforeCreate + 1);
-        System.out.println(visitDao.findAll().size());
         assertThat(visitDao.findAll()).containsExactlyInAnyOrder(visit1, visit2, testVisit);
+    }
+
+    @Test
+    public void create_visitWithNullDate() throws Exception {
+        Forest testForest = new Forest();
+        testForest.setName("Test Forest");
+
+        MushroomHunter testHunter = new MushroomHunter();
+        testHunter.setFirstName("Maradona");
+        testHunter.setSurname("Sandokan");
+        testHunter.setUserNickname("bidG");
+
+        em.persist(testForest);
+        em.persist(testHunter);
+
+        Visit testVisit = new Visit();
+        testVisit.setHunter(testHunter);
+        testVisit.setForest(testForest);
+        testVisit.setDate(null);
+
+        assertThatThrownBy(() -> visitDao.create(testVisit)).isInstanceOf(ConstraintViolationException.class);
+    }
+
+    @Test
+    public void update_validVisitDate() {
+        LocalDate newDate = LocalDate.of(2017,5,12);
+        Visit foundVisit = em.find(Visit.class, visit1.getId());
+
+        foundVisit.setDate(newDate);
+        visitDao.update(foundVisit);
+        em.flush();
+
+        assertThat(em.find(Visit.class, visit1.getId()).getDate()).isEqualTo(newDate);
+    }
+
+    @Test
+    public void update_validVisitForestName() {
+        String forestNewName = "Sexy forest";
+        Forest foudForest = em.find(Forest.class, forest1.getId());
+        Visit foundVisit = em.find(Visit.class, visit1.getId());
+
+        foudForest.setName(forestNewName);
+
```

Comment:
I don't know, but isn't a forest.update() missing? And then, wouldn't it be too many tests in one @Test (visit and forest too)?